app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/ExposureNotificationsRepository.kt

```diff
@@ -311,14 +311,41 @@ class ExposureNotificationsRepository @Inject constructor(
                 accepted = false
             )
         })
-        val latestExposure = db.dao().getLatest().firstOrNull()?.daysSinceEpoch
-        val lastNotifiedExposure = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
-        if (latestExposure != null && latestExposure != lastNotifiedExposure) {
+
+        // latest exposure found in the database
+        val latestExposure = db.dao().getLatest().firstOrNull()
+        val latestExposureTime = latestExposure?.daysSinceEpoch
+
+        // latest exposure that the user was not notified about yet
+        val lastNotifiedExposureTime = db.dao().getLastNotified().firstOrNull()?.daysSinceEpoch
+
+        // the app should show a notification if there is a new exposure the user was not notified
+        // about, yet, or if there is an exposure, but the app has not been opened since the last
+        // last notification
+        val newExposureFound = latestExposureTime != lastNotifiedExposureTime
+        val userNotNotifiedAboutLatest = latestExposureTime != null && newExposureFound
```

Comment:
Tahle podmínka mi nějak nedává smysl. Proč je potřeba? Proč když je nová exposure a poslední exposure existuje, ještě o ní nebyl user notifikován?

Reply:
nova a posledni exposure je ta ista vec. iba som sa snazila rozdelit ten kod pretoze tie podmienky maju viac ako 120 znakov. mozem ale pridat ten null check na riadok 325. 
ale inak - user o nej nebol notifikovany, pretoze prave o nej ide byt notifikovany, pretoze o par riadkov vyssie sa este len ulozila do DB. Imho to je kod, ktory tam uz bol a ja som ho akurat zrefaktorovala.